docs/19_security_audit_document.md

```diff
@@ -0,0 +1,258 @@
+| `list`    | `Prohibited`                                                  |
+| `delete`  | `Prohibited`                                                  |
+
+### The `allowed_email_domains` collection
+
+The `allowed_email_domains` collection contains the allowed email domains for the `Google Sign-In` method. The single document of this collection stands for the email domain that is allowed and may be used to authorize a user.
+
+The documents of this collection do not have any fields. Instead, the ID of the document represents a single email domain. This prevents duplicates of email domains and simplifies validating as documents aren't to be fetched.
+
+Here is a table of security rules applied to the `allowed_email_domains` collection.
+
+| Operation       | [Security Rules](#security-rules-description) |
+|-----------------|----------------|
+| `read`, `write` | `Prohibited`   |
+
+### The `feature_config` collection
+
+The `feature_config` collection contains the configuration values to enable or disable the application features. This collection contains a single `feature_config` document containing all configurations for the whole application.
+
+Consider the following table that describes the fields of the `feature_config` document:
+
+| Field                           | Description                                                           |
+|---------------------------------|-----------------------------------------------------------------------|
+| `isDebugMenuEnabled`            | Indicates whether the `Debug Menu` feature is enabled.                |
+| `isPasswordSignInOptionEnabled` | Indicates whether the `Email and Password` sign in option is enabled. |
+
+Here is a table of security rules applied to the `feature_config` collection.
+
+| Operation | [Security Rules](#security-rules-description) |
+|-----------|----------------|
+| `read`    | `Allowed`      |
+| `write`   | `Prohibited`   |
+
+### Security Rules Description
+
+| Rule                 | Description |
+|----------------------|-------------|
+| `isAccessAuthorized` | The user is authenticated and the email domain is valid. |
+| `isProjectValid`     | A name of the given project is a string and the request data contains only the project name. |
+| `isProjectGroupValid`| A name of the given project group is a string having maximum 255 characters, the given project IDs is a list containing no more than 20 project IDs, and the request data contains only the project group name and the projects' IDs. |
+| `isBuildValid`       | A project ID of the given build is a string and this project exists in the database, the build number is an integer, the start time of the build is a timestamp, the build status is either successful, failed, or unknown, the build duration is an integer, the workflow name is either a string or a null, the url of the build is a string, the test coverage is either float or null. |
```

Comment:
```suggestion
| `isBuildValid`       | A project with the project ID from the request data exists in the database, and all the rest fields have a valid data type. The given build data contains only allowed fields declared in the [build collection](#the-build-collection) section |
```

Would it be a bit better to avoid mentioning the concrete data types here?